db/queries/scheduled_services.sql

```diff
@@ -0,0 +1,81 @@
+-- name: InsertScheduledService :one
+INSERT INTO scheduled_service
+    (id, system_pk, monday, tuesday, wednesday, thursday, friday, saturday, sunday, start_date, end_date, feed_pk)
+VALUES
+    (sqlc.arg(id), sqlc.arg(system_pk), sqlc.arg(monday), sqlc.arg(tuesday), sqlc.arg(wednesday), sqlc.arg(thursday), sqlc.arg(friday), sqlc.arg(saturday), sqlc.arg(sunday), sqlc.arg(start_date), sqlc.arg(end_date), sqlc.arg(feed_pk))
+RETURNING pk;
+
+-- name: UpdateScheduledService :exec
+UPDATE scheduled_service SET
+    system_pk = sqlc.arg(system_pk),
+    monday = sqlc.arg(monday),
+    tuesday = sqlc.arg(tuesday),
+    wednesday = sqlc.arg(wednesday),
+    thursday = sqlc.arg(thursday),
+    friday = sqlc.arg(friday),
+    saturday = sqlc.arg(saturday),
+    sunday = sqlc.arg(sunday),
+    start_date = sqlc.arg(start_date),
+    end_date = sqlc.arg(end_date),
+    feed_pk = sqlc.arg(feed_pk)
+WHERE pk = sqlc.arg(pk);
+
+-- name: InsertScheduledServiceAddition :exec
+INSERT INTO scheduled_service_addition
+    (service_pk, date)
+VALUES
+    (sqlc.arg(service_pk), sqlc.arg(date));
+
+-- name: DeleteScheduledServiceAdditions :exec
+DELETE FROM scheduled_service_addition
+USING scheduled_service
+WHERE scheduled_service.pk = scheduled_service_addition.service_pk
+AND feed_pk = sqlc.arg(feed_pk);
```

Comment:
I wonder does `ON DELETE CASCADE` work for deleting these entries?

Reply:
I think it should, removed this in latest update and tests still pass from a cascading delete (the test "delete stale service additions and removals" should hopefully cover this).